src/utils/validators.js

```diff
@@ -0,0 +1,305 @@
+ * @param {number} [opts.minLength] The minimum string length.
+ * @param {number} [opts.maxLength] The maximum string length.
+ *
+ * @throws {TypeError} Value should be a valid string.
+ */
+export function validateString ({ value, fieldName = '', minLength, maxLength }) {
+  if (!BaseOperation.isValidString(value, minLength, maxLength)) {
+    throw new TypeError(
+      `${fieldName} must be a valid string with params: ` +
+      `${JSON.stringify({ minLength, maxLength })}, got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/** Keys validators */
+
+/**
+ * Validate value to be a public key.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid public key.
+ */
+export function validatePublicKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidPublicKey(value)) {
+    throw new TypeError(
+      `${fieldName} must be a valid public key, got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/**
+ * Validate value to be a secret key.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid secret key.
+ */
+export function validateSecretKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidSecretKey(value)) {
+    throw new TypeError(
+      `${fieldName} must be a valid secret key, got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/**
+ * Validate value to be a balance key.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid balance key.
+ */
+export function validateBalanceKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidBalanceKey(value)) {
+    throw new TypeError(
+      `${fieldName} must be a valid balance key, got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/** Fields validators */
+
+/**
+ * Validate value to be an int64 amount.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ * @param {boolean} [opts.allowZero] Is zero amount allowed.
+ * @param {number} [opts.min] The minimum amount.
+ * @param {number} [opts.max] The maximum amount.
+ * @param {number} [opts.maxDecimalPlaces] The maximum decimal places allowed.
+ *
+ * @throws {TypeError} Value should be a valid int64 amount.
+ */
+export function validateAmount ({
+  value,
+  fieldName = '',
+  allowZero,
+  max,
+  min,
+  maxDecimalPlaces
+}) {
+  if (!BaseOperation.isValidAmount(value, allowZero, max, min, maxDecimalPlaces)) {
+    throw new TypeError(
+      `${fieldName} must be a valid int64 string with params:` +
+      `${JSON.stringify({ min, max, allowZero, maxDecimalPlaces })}, ` +
```

Comment:
let’s introduce a function that will do the stringification, but also will omit `undefined` fields. no more other transformations like you have before is not needed

Reply:
Discussed IRL